core/src/main/java/com/scalar/db/transaction/consensuscommit/RecoveryHandler.java

```diff
@@ -42,70 +37,22 @@ public RecoveryHandler(
   public void recover(Selection selection, TransactionResult result) {
     logger.debug("recovering for {}", result.getId());
 
-    // as the result doesn't have before image columns, need to get the latest result
-    Optional<TransactionResult> latestResult;
-    try {
-      latestResult = getLatestResult(selection, result);
-    } catch (ExecutionException e) {
-      logger.warn("can't get the latest result", e);
-      return;
-    }
-
-    if (!latestResult.isPresent()) {
-      // indicates the record has been deleted by another transaction
-      return;
-    }
-
-    if (latestResult.get().isCommitted()) {
-      // indicates the record has been committed by another transaction
-      return;
-    }
```

Comment:
Removed this logic to get the latest result here because we need to do that only when rolling back a record. We will get the latest result in `RollbackMutationComposer`.
https://github.com/scalar-labs/scalardb/pull/681/files#diff-93002106f4ee40479af069dedcc4f0c5c392a6812b5da81363e80d5f0f530f3aR50-R53